src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+            logDates.forEach((logDay) =>{
+                const entriesForDate = bpDataPoints.filter( a => a.dateLabel == logDay);
+                const systolicEntriesForDate = entriesForDate.map( e => e.systolic);
+                const systolicInterval = buildInterval(systolicEntriesForDate);
+                const systolicAverage = systolicEntriesForDate.reduce((a, b) => a + b) / systolicEntriesForDate.length;
+                const diastolicEntriesForDate = entriesForDate.map( e => e.diastolic);
+                const diastolicInterval = buildInterval(diastolicEntriesForDate);
+                const diastolicAverage = diastolicEntriesForDate.reduce((a, b) => a + b) / diastolicEntriesForDate.length;
+                const dataPoint : DataPoint = {dataSet1: diastolicInterval, dataSet2: systolicInterval};
+                var db : Dumbbell = {dataPoint  :dataPoint, xValue : logDay, class : assignClass(systolicAverage, diastolicAverage)};
+                dbs.push({date: entriesForDate[0].date, dumbbell: db, averageSystolic : systolicAverage, averageDiastolic : diastolicAverage});
+            });
+        }
+
+        return dbs;
+    }
+
+    function pageWeeklyData(startOfWeek: Date, freshData? : BloodPressureDumbbell[]){
+        const useData = freshData ?? bpData;
+        const weekData : Dumbbell[] = [];
+        const weekDataForMetrics : BloodPressureDumbbell[] = [];
+        for (let i = 0; i < 7; i++) {
+            var currentDate = startOfDay(addDays(startOfWeek, i));
+            var dataForDay = useData.find( db => isEqual(db.date, currentDate));
+            if (!dataForDay)
+            {
+                var formattedDate = format(currentDate, "MM/dd");
+                weekData.push({xValue : formattedDate});
+            } 
+            else 
+            {
+                weekDataForMetrics.push(dataForDay);
+                weekData.push(dataForDay.dumbbell);
+            }
+        }
+
+        const weeklyMetrics = getWeeklyAggregates(weekDataForMetrics);
+        setMetrics(weeklyMetrics);
+        setWeeklyDataForViz(weekData);
+        setStartOfWeek(startOfWeek);
+    }
+
+    function getWeeklyAggregates(dbs : BloodPressureDumbbell[]){
+        if (dbs.length === 0) return;
+
+        let diastolicReadings = dbs.map(db => db.averageDiastolic);
+        let diastolicWeeklyAvg = diastolicReadings.reduce((a, b) => a + b) / diastolicReadings.length;
+        diastolicWeeklyAvg = Math.round(diastolicWeeklyAvg);
+        let diastolicWeeklyAvgAlert = getDiastolicCategory(diastolicWeeklyAvg);
+
+        let systolicReadings = dbs.map(db => db.averageSystolic);
+        let systolicWeeklyAvg = systolicReadings.reduce((a, b) => a + b) / systolicReadings.length;
+        systolicWeeklyAvg = Math.round(systolicWeeklyAvg);
+        let systolicWeeklyAvgAlert = getSystolicCategory(systolicWeeklyAvg);
+
+        let bpMetrics : BloodPressureMetrics = {
+            averageSystolic : systolicWeeklyAvg, 
```

Comment:
Aren't you averaging an average here?  Is that what is intended?  e.g. What if someone took 20 readings on day 1 and then 1 reading on day 2...is it supposed to be the average of all 21 readings

Reply:
Yes, averaging the daily averages where a daily average exist.
In the example you provided, (Day1Average + Day2Average)/totalDaysWhereDataExists